### DISCUSSION TITLE
Two-factor authentication, change mobile device.

### DISCUSSION THREAD

**Discussion by stanislavs-flutterint** [community/community#89803] [New to GitHub] [answered]
I am planning to change phone that was used to setup Two-factor authentication.
Unfortunately, documentation does not have any information on what one should do in this scenario.

So, how could I enable Authy to generate correct Github code on new mobile device? 
Thank you.

**Comment by fury-05** [comment, marked as answer]
I assume that you have Authy as authentication for 2fa, if so follow the following steps:

1. open Authy in your old phone.
2. click on 3 dots
3. go to settings
4. click on devices tab
5. turn on allow multi-device
6. then turn on use existing device to approve a new device.
7. on you new phone install and open Authy and follow same process but use same credentials.
8. it will ask for otp use old phone and add code generated in your old phone after entering it you will be able to use new device with all added accounts.

Hope you are looking to this solution let me know if it resolved your query.

**Reply by kulapati** [reply]
I am using microsoft's authentication app and there is no devices tab . What should I do?

**Reply by DNin01** [reply]
> I am using microsoft's authentication app and there is no devices tab . What should I do?

You'll have to turn on cloud backup and then restore your credentials during the authenticator setup process on your new device.

**Reply by kulapati** [reply]
Thank you
